### PR TITLE
Add new Customer model (including import script)

### PR DESCRIPTION
**Pull request by tekin**
This adds the Customer model and a script to import all the customers we currently know about so that we can use the data to generate the finance report. More details in the commit messages.

### CODE REVIEW

File: db/migrate/20180807093650_create_customers.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    create_table :customers, id: :uuid do |t|
+      t.string :name, null: false
+      t.string :postcode
+      t.integer :urn, null: false
+      t.integer :sector, null: false
+
+      t.timestamps
+    end
+
+    add_index :customers, :urn, unique: true
```
Comment (leeky) [inline db/migrate/20180807093650_create_customers.rb]: :+1:

File: app/models/customer.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Customer < ApplicationRecord
+  enum sector: %w[central_government wider_public_sector]
```
Comment (leeky) [inline app/models/customer.rb]: Is it worth being explicit with the mappings here?

File: spec/models/customer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe Customer do
+  it { is_expected.to validate_presence_of(:name) }
+  it { is_expected.to validate_presence_of(:urn) }
+  it { is_expected.to validate_presence_of(:sector) }
```
Comment (leeky) [inline spec/models/customer_spec.rb]: How about a `validate_uniqueness_of(:urn)` test here for completeness?

File: db/data_migrate/20180807095900_import_customers.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require 'csv'
+require 'progress_bar'
+
+puts 'Reading CSV file...'
+csv = CSV.table(Rails.root.join('db', 'data_migrate', '20180807095900_customers.csv'))
+
+puts 'Importing customers...'
+bar = ProgressBar.new(csv.count)
```
Comment (leeky) [inline db/data_migrate/20180807095900_import_customers.rb]: Does this actually render properly inside the docker containers slightly-broken terminal?

File: db/data_migrate/20180807095900_import_customers.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+csv.each do |customer_row|
+  postcode = customer_row[:postcode] == 'XXXX' ? nil : customer_row[:postcode]
+
+  Customer.create!(
```
Comment (leeky) [inline db/data_migrate/20180807095900_import_customers.rb]: How about making this able to be run more than once? (i.e. if the VPN dies in the middle of the script running)

Also, it's probably worth calling `strip` on all these input strings, and a lot of the other exports have included erroneous whitespace.

File: db/data_migrate/20180807095900_import_customers.rb (hunk 4 above)
Comment (tekin) [reply db/data_migrate/20180807095900_import_customers.rb]: tested locally and it appears it does.

File: app/models/customer.rb (hunk 2 above)
Comment (tekin) [reply app/models/customer.rb]: updated.

File: spec/models/customer_spec.rb (hunk 3 above)
Comment (tekin) [reply spec/models/customer_spec.rb]: adding for completeness, although I'll be honest I'm not a huge fan of these declarative specs. not sure what value they add as they are simply mirror the validations on the model.

File: db/data_migrate/20180807095900_import_customers.rb (hunk 5 above)
Comment (tekin) [reply db/data_migrate/20180807095900_import_customers.rb]: updated.

**Comment by leeky** [comment]
:+1:

**Pull request merged**